can-auth/vatican.c

```diff
@@ -199,9 +232,35 @@ int VULCAN_FUNC vulcan_init(ican_t *ican, ican_link_info_t connections[],
     ASSERT(rv >= 0);
     pr_info("CAN controller initialized");
 
+    #if VATITACAN
+        // Calculate mask for IAT nonce construction
+    	nonce_size = VATITACAN_NONCE_SIZE;
+	while (nonce_size)
+	{
+	    nonce_mask = nonce_mask*2;
+	    nonce_size--;
+	}
+	nonce_mask--;
+
+	#if defined(VULCAN_SM)
+	    // Check for IAT buffer and index lying outside of PM
+            ASSERT(sancus_is_outside_sm(VULCAN_SM, can_iat_timings, CAN_IAT_BUFFER_SIZE));
```

Comment:
nice! In principle, this is not strictly needed when the pointer is static and not passed in at runtime (as I understood your prior implementation did). As long as the pointer value is hard-coded in the enclave text section, eg when it's the address of a global var in C, the remote stakeholder can use remote attestation to verify that it indeed lies outside the enclave. However, it doesn't hurt to have it here as well and it's quite clean ^^ You might want to add as well then a check for the iat_index global var (cf further)